StortSpelProjekt/Game/src/Gamefiles/Shop.cpp

```diff
@@ -29,6 +30,23 @@ void Shop::RandomizeInventory()
 	int upgradeNum;
 	std::string name;
 	bool inStock = false;
+
+	// How many upgrades have reached max level
+	int maxLevelUpgrades = 0;
+	for (auto u : m_AllAvailableUpgrades)
+	{
+		if (u.second->GetLevel() == u.second->GetMaxLevel())
+		{
+			maxLevelUpgrades++;
+		}
+	}
+	// If we dont have enough upgrades that are not max level to fill inventory to InvSize
+	// then we lower InvSize
```

Comment:
This comment tries to explain that "If an upgrade is at max level, make it unavailable for purchase"? Please clarify, I got confused by it.